From 61e2e0e4b0d0e526197d28b982ac6478ebc45fbd Mon Sep 17 00:00:00 2001
From: Noam Camus <ncamus@habana.ai>
Date: Thu, 19 Mar 2020 00:54:01 +0200
Subject: [PATCH] [SW-0] goya2: arc: add spi_ssi_cs0_get()

Override weak definition with this dedicated routine which match goya2
architecture

Change-Id: Icdb567529f37d96bfde52ec5d3d3561d17b0a782
---

diff --git a/goya2_arc/common.c b/goya2_arc/common.c
index 6ee4280..e713727 100644
--- a/goya2_arc/common.c
+++ b/goya2_arc/common.c
@@ -7,6 +7,7 @@
 
 #include <pcie_phy_regs.h>
 #include <psoc_global_conf_regs.h>
+#include <psoc_gpio1_regs.h>
 #include <psoc_qspi_regs.h>
 #include <pcie_aux_regs.h>
 #include <pcie_wrap_regs.h>
@@ -219,6 +220,14 @@
 	init_prstn();
 }
 
+void spi_ssi_cs0_get(unsigned int *mask, unsigned int *data_offset,
+		     unsigned int *out_en_offset)
+{
+	*mask = BIT(19 % 16);
+	*data_offset = mmPSOC_GPIO1_DATAOUT;
+	*out_en_offset = mmPSOC_GPIO1_OUTENSET;
+}
+
 void spi_base_get(unsigned int *regs)
 {
 	*regs = mmPSOC_QSPI_CTRLR0;
